src/System.Management.Automation/engine/hostifaces/PSTask.cs

```diff
@@ -672,46 +681,21 @@ public void Dispose()
         /// This method is not multi-thread safe and assumes only one thread waits and adds tasks.
         /// </summary>
         /// <param name="task">Task to be added to pool.</param>
-        /// <param name="dataStreamWriter">Optional cmdlet data stream writer.</param>
         /// <returns>True when task is successfully added.</returns>
-        public bool Add(
-            PSTaskBase task, 
-            PSTaskDataStreamWriter dataStreamWriter = null)
+        public bool Add(PSTaskBase task)
         {
             if (!_isOpen)
             {
                 return false;
             }
 
-            WaitHandle[] waitHandles;
-            if (dataStreamWriter != null)
-            {
-                waitHandles = new WaitHandle[]
-                {
-                    _addAvailable,                          // index 0
-                    _stopAll,                               // index 1
-                    dataStreamWriter.DataAddedWaitHandle    // index 2
-                };
-            }
-            else
-            {
-                waitHandles = new WaitHandle[]
-                {
-                    _addAvailable,                          // index 0
-                    _stopAll,                               // index 1
-                };
-            }
+            // Block until either room is available, or a stop is commanded
```

Comment:
Does `room` here mean space in the queue?

Reply:
Yes.  Will fix.